pkg/common-controller/snapshot_controller.go

```diff
@@ -266,8 +266,10 @@ func (ctrl *csiSnapshotCommonController) checkandRemoveSnapshotFinalizersAndChec
 		}
 
 		if !inUse {
-			klog.V(5).Infof("checkandRemoveSnapshotFinalizersAndCheckandDeleteContent: Set VolumeSnapshotBeingDeleted annotation on the content [%s]", content.Name)
-			ctrl.setAnnVolumeSnapshotBeingDeleted(content)
+			if content != nil {
+				klog.V(5).Infof("checkandRemoveSnapshotFinalizersAndCheckandDeleteContent: Set VolumeSnapshotBeingDeleted annotation on the content [%s]", content.Name)
+				ctrl.setAnnVolumeSnapshotBeingDeleted(content)
```

Comment:
This is called in multiple places.  Can you add a nil check inside setAnnVolumeSnapshotBeingDeleted()?
I actually just ran into this yesterday so I added it here:
https://github.com/kubernetes-csi/external-snapshotter/pull/261/files#diff-214a2fc68717c170415a3578536d3b74R1281

I'll remove my change once your PR is merged.

Reply:
I've updated this to also include the check inside of `setAnnVolumeSnapshotBeingDeleted()`. I was hitting the panic on line 270 above, when we were printing out the debug statement.

I can remove the name of the content, since it should print this out within `setAnnVolumeSnapshotBeingDeleted` immediately after it checks for nil now.

Reply:
This nil check on line 269 can be removed now since you moved the check inside setAnnVolumeSnapshotBeingDeleted()

Reply:
I removed the nil check here, but I also updated the debug logging to not print the content name. That's where I was hitting the panic